PEP 784: Mark as Accepted (#4387)
python/peps: peps/pep-0784.rst

Author: Emma Harper Smith <emma@python.org>
Sponsor: Gregory P. Smith <greg@krypto.org>
Discussions-To: https://discuss.python.org/t/87377
Status: Accepted
Type: Standards Track
Created: 06-Apr-2025
Python-Version: 3.14
Post-History:
  `07-Apr-2025 <https://discuss.python.org/t/87377>`__,

> Resolution: `25-Apr-2025 <https://discuss.python.org/t/pep-784-adding-zstandard-to-the-standard-library/87377/138>`__

Reviewer: ```suggestion
Resolution: `25-Apr-2025 <https://discuss.python.org/t/87377/138>`__
```

Author: TIL, thanks!

PyPI. To avoid breaking users of one of the existing bindings, this PEP
proposes introducing a new namespace for compression libraries,
``compression``. This name is already reserved on PyPI for use in the
standard library. The new Zstandard module will be named ``compression.zstd``.

> Other compression modules will be re-exported to the ``compression`` namespace.

Reviewer 2: ```suggestion
Other compression modules will be re-exported in the new ``compression`` package.
```

selected due to the current contents of the module being I/O related helpers
for stream APIs (e.g. ``LZMAFile``) in standard library compression modules.

Backwards Compatibility
=======================

This PEP introduces no backwards incompatible changes. There are currently no
plans to deprecate or remove the existing compression modules. Any deprecation

> or removal of the existing modules is left to a future decision.

Reviewer: Is it worth commenting that the expectation of a future PEP is a timeline of at least 5-10 years? Currently it says nothing, which readers may interpret as 'the old names could be at risk at any point'.

Reviewer 2: It could say "... is left to a future decision, but no sooner than 5 years".  Let's see what other SC members think.

Author: I almost put a reference to the default deprecation timeline of 5 years but I like the framing of "no sooner than 5 years"

Reviewer 3: "no sooner than" sounds good to me.  The thing to communicate is that there is no breaking change people need to plan for in this PEP and if we ever decide to, it won't be soon.

Author: I added this wording in 2ce131663f331ff3fccba08152ab9b104d3b6551.

=================

Documentation for the new module is in the reference implementation branch. The
documentation for existing modules will be updated to reference the new names

> as well.

Reviewer: Do the new names now become canonical? The PEP should take a position.

Reviewer 2: I think yes the new names should become canonical.  Let's see what other SC members think.

Reviewer 3: The intent is for them to become canonical as that is why the compression namespaces is useful.  I expect we may iterate on how exactly to present this in our docs outside the PEP. 

As far as the text in the PEP goes adding something along the lines of:

"The use of the new compression names will be promoted over the original top level module names in the Python documentation when the users minimum Python version support requirements makes that feasible."

Author: I added a sentence about this in 53ab0b58. Not sure if I should be stronger in my wording but hopefully that clearly makes a stance.

Author: Oh, my page was open from last night and I missed Greg's comment. I'll include that text.

Author: Added in 9d58ebce, I made a slight edit to the sentence to clarify, hopefully that's ok!
